network/p2p/middleware/middleware.go

```diff
@@ -347,7 +347,7 @@ func (m *Middleware) topologyPeers() peer.IDSlice {
 }
 
 // OnNodeBlockListUpdate removes all peers in the blocklist from the underlying libp2pnode.
-func (m *Middleware) OnNodeBlockListUpdate(blockList flow.IdentifierList) {
+func (m *Middleware) OnNodeDisallowListUpdate(blockList flow.IdentifierList) {
```

Comment:
note that your implementation violates the API specification: https://github.com/onflow/flow-go/blob/9a82acd287ba7f35e143de182fc3d556632325fb/network/p2p/consumer.go#L16-L18

Reply:
You are right. The issue was due to both the consumer and distributor would share the same interface while the distributor was expected to be non-blocking and the consumer on the other hand could be blocking. To disambiguate the semantics, I decoupled the interface for distributor and consumer: 

https://github.com/onflow/flow-go/blob/d7387276e95ddbedae02320cac68816e03e96d1b/network/p2p/consumer.go#L36-L53